archunit/src/main/java/com/tngtech/archunit/core/domain/JavaPackage.java

```diff
@@ -227,32 +322,32 @@ public Set<JavaClass> getAllClasses() {
     }
 
     /**
-     * @return all (direct) sub-packages contained in this package, e.g. {@code [java.lang, java.io, ...]} for package {@code java}
-     * (compare {@link #getAllSubpackages()})
+     * @return all (direct) subpackages contained in this package, e.g. for package {@code java} this would be
+     *         {@code [java.lang, java.io, ...]} (compare {@link #getSubpackageTree()})
      */
     @PublicAPI(usage = ACCESS)
     public Set<JavaPackage> getSubpackages() {
         return ImmutableSet.copyOf(subpackages.values());
     }
 
     /**
-     * @return all sub-packages including nested sub-packages contained in this package,
-     * e.g. {@code [java.lang, java.lang.annotation, java.util, java.util.concurrent, ...]} for package {@code java}
-     * (compare {@link #getSubpackages()})
+     * @return all subpackages contained in the package tree of this package. I.e. all subpackages, subpackages
+     *         of subpackages, and so on. For package {@code java} this would be
+     *         {@code [java.lang, java.lang.annotation, java.util, java.util.concurrent, ...]} (compare {@link #getSubpackages()})
      */
     @PublicAPI(usage = ACCESS)
-    public Set<JavaPackage> getAllSubpackages() {
+    public Set<JavaPackage> getSubpackageTree() {
```

Comment:
Naming - my first thought when I read `getSubpackageTree` is "this returns a tree-like data structure", while it actually returns a `Set`.
My second thought is "ah, right, that's okay :)"

A few alternative options I can think of (not sure better or not):
- `getSupbackagesInTree`
- `getSupbackagesInPackageTree`
- `getSubpackagesRecursively`

Reply:
Oh yeah, now that you say it I completely agree! I'm sold on `getSubpackagesInTree` :slightly_smiling_face: Because "subpackages" already sets the context, I think just using "tree" is fine then (I don't think there is any ambiguity; what other "tree" would I get my subpackages from :man_shrugging:)
I mean, likely `getSubpackagesRecursively` is the most precise one, but given that we now set a strong connotation with "package tree" I think it makes sense to still use "tree" here...